flang/lib/Optimizer/CodeGen/TypeConverter.h

```diff
@@ -141,6 +141,24 @@ class LLVMTypeConverter : public mlir::LLVMTypeConverter {
     parts.push_back(getDescFieldTypeModel<4>()(&getContext()));
     parts.push_back(getDescFieldTypeModel<5>()(&getContext()));
     parts.push_back(getDescFieldTypeModel<6>()(&getContext()));
+    if (rank == -1) {
```

Comment:
Magic number.

Reply:
Yes, that was the default argument. I added a constexpr `unknownRank()` helper to make that a bit cleaner.